### PR TITLE
Removed redundant json sanitizing from BraveVPNService

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/23726

As BraveVPNService gets json via APIRequestHelper,
doing json sanitizing from it is redundant.
APIRequestHelper already does it for client.

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`npm run test brave_unit_tests -- --filter=*ResponseSanitizingTest`

### CODE REVIEW

**Review by spylogsster — APPROVED** [review]
LGTM

File: components/brave_vpn/brave_vpn_service.cc [hunk 1]
```diff
@@ -1216,38 +1216,8 @@ void BraveVpnService::OnGetResponse(
...
-  }
-
-  if (success && !sanitized_json_response.value.has_value()) {
-    VLOG(1) << "Empty response";
-    success = false;
-  }
-
   if (success)
-    json_response = sanitized_json_response.value.value();
+    json_response = body;
```
[the remark this answers is not in the document]
Comment (simonhong) [reply components/brave_vpn/brave_vpn_service.cc]: Done.

File: components/brave_vpn/brave_vpn_service.cc [hunk 2]
```diff
@@ -1202,41 +1202,8 @@ void BraveVpnService::OnGetResponse(
     int status,
     const std::string& body,
     const base::flat_map<std::string, std::string>& headers) {
-  std::string json_response;
```
Comment (bsclifton) [inline components/brave_vpn/brave_vpn_service.cc:1205]: With this code gone, maybe we can have a comment in `BraveVpnService::OAuthRequest` with something like:
```
// NOTE: api_request_helper_ uses JsonSanitizer to sanitize input made with requests
```

**Review by bsclifton — APPROVED** [review]
Left one comment; both otherwise, LGTM! Thanks for adding a test 😄 

If we wanted to get crazy, a better test would be to have an interface for `api_request_helper_` that we can make a mock implementation for... to ensure it gets called. But I think that the check for handling invalid input is good 👍

File: components/brave_vpn/brave_vpn_service.cc (hunk 2 above)
Comment (simonhong) [reply components/brave_vpn/brave_vpn_service.cc:1205]: Added 👍🏼

**Pull request merged**